tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,776 @@
+    if(system('tar', 'zcvf', $filesBackupPath, '--absolute-names', '--files-from', $tmpFileName)) {
+        print STDERR "backup command failed: $!\n";
+        exit $NeuroDB::ExitCodes::PROGRAM_EXECUTION_FAILURE;
+    } 
+
+    print "File $filesBackupPath successfully created.\n";
+}
+
+=pod
+
+=head3 deleteUploadsInDatabase($dbh, $uploadsRef, $filesRef)
+
+This method deletes all information in the database associated to the given upload(s). More specifically, it 
+deletes records from tables C<notification_spool>, C<tarchive_files>, C<tarchive_series>, C<files_intermediary>,
+C<parameter_file>, C<files>, C<mri_protocol_violated_scans>, C<mri_violations_log>, C<MRICandidateErrors>
+C<mri_upload> and C<tarchive>. It will also set the C<Scan_done> value of the scan's session to 'N' for each upload
+that is the last upload tied to that session. All the delete/update operations are done inside a single transaction so 
+either they all succeed or they all fail (and a rollback is performed).
+
+INPUTS:
+  - $dbh       : database handle.
+  - $uploadsRef: reference on a hash of hashes containing the uploads to delete. Accessed like this:
+                 C<< $uploadsRef->{'1002'}->{'TarchiveID'} >>(this would return the C<TarchiveID> of the C<mri_upload>
+                 with ID 1002). The properties stored for each hash are: C<UploadID>, C<TarchiveID>, C<ArchiveLocation>
+                 and C<SessionID>.
+  - $filesRef: reference to the array that contains the file informations for all the files
+  that are associated to the upload(s) passed on the command line.
+                  
+=cut
+sub deleteUploadsInDatabase {
+    my($dbh, $uploadsRef, $filesRef)= @_;
+    
+    $dbh->{'AutoCommit'} = 0;
+    
+    my $query = "DELETE FROM notification_spool WHERE ProcessID IN ("
+              . join(',', ('?') x keys %$uploadsRef)
+              . ")";
+    $dbh->do($query, undef, keys %$uploadsRef);
+    
+    my $tarchiveID = $filesRef->{'tarchive'}->[0]->{'TarchiveID'};
+    
+    $query = "DELETE FROM tarchive_files WHERE TarchiveID = ?";
+    $dbh->do($query, undef, $tarchiveID);
+    
+    $query = "DELETE FROM tarchive_series WHERE TarchiveID = ?";
+    $dbh->do($query, undef, $tarchiveID);
+    
+    if(@{ $filesRef->{'parameter_file'} }) {
+        my @parameterFileIDs = map { $_->{'FileID'} } @{ $filesRef->{'parameter_file'} };
+        $query  = sprintf(
+            "DELETE FROM parameter_file WHERE FileID IN (%s)",
+            join(',', ('?') x @parameterFileIDs)
+        );
+        $dbh->do($query, undef, @parameterFileIDs);
+    }
+
+    my @intermediaryFileIDs = map { $_->{'IntermedID'} } @{ $filesRef->{'files_intermediary'} };
+    if(@intermediaryFileIDs) {
+        $query = sprintf(
+            'DELETE FROM files_intermediary WHERE IntermedID IN (%s)', 
+            join(',', ('?') x @intermediaryFileIDs)
+        );
+        $dbh->do($query, undef, @intermediaryFileIDs);     
+
```

Comment:
delete from parameter_file before deleting from files and it should be good to go!

Reply:
Done.